src/CommandHandler/Wishlist/AddProductToWishlistHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusWishlistPlugin\CommandHandler\Wishlist;
+
+use BitBag\SyliusWishlistPlugin\Command\Wishlist\AddProductToWishlist;
+use BitBag\SyliusWishlistPlugin\Entity\WishlistInterface;
+use BitBag\SyliusWishlistPlugin\Exception\ProductNotFoundException;
+use BitBag\SyliusWishlistPlugin\Factory\WishlistProductFactoryInterface;
+use BitBag\SyliusWishlistPlugin\Repository\WishlistRepositoryInterface;
+use BitBag\SyliusWishlistPlugin\Updater\WishlistUpdaterInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+class AddProductToWishlistHandler implements MessageHandlerInterface
+{
+    private WishlistProductFactoryInterface $wishlistProductFactory;
+
+    private WishlistUpdaterInterface $wishlistUpdater;
+
+    private ProductRepositoryInterface $productRepository;
+
+    private WishlistRepositoryInterface $wishlistRepository;
+
+    public function __construct(
+        WishlistProductFactoryInterface $wishlistProductFactory,
+        WishlistRepositoryInterface $wishlistRepository,
+        WishlistUpdaterInterface $wishlistUpdater,
+        ProductRepositoryInterface $productRepository
+    )
+    {
```

Comment:
I think that those two should be in one line, like ) {

Reply:
This is still WIP :) we will use ECS at the end.